components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -313,6 +315,35 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
             //     .unwrap_or_else(move |e| fail_f(e,
             // DeleteStrategy::DeleteBlobs));
         }
+
+        self.ensure_write_buffer();
+        // Delete range may not fill anything in memtable if the range is fit well with
+        // sst files. If no further write/delete is performaned for this cf, flushed
+        // index and modified index in apply trace will never matched, which makes admin
+        // flush index never progress. It severely impacts raft log gc and raft log
+        // replay.
+        // So artificially add a magic delete here to make at least one element will be
+        // put in memtable.
+        let res = if cf.is_empty() || cf == CF_DEFAULT {
+            self.write_batch
+                .as_mut()
+                .unwrap()
+                .delete(MAGIC_KEY.as_bytes())
```

Comment:
Instead of using MAGIC_Key, we can simply delete start_key as it won't have correctness issue. 
Here MAGIC_KEY is kind of ugly and it may lead to problems.

Reply:
And why not use delete_cf for both branch?

Reply:
> Instead of using MAGIC_Key, we can simply delete start_key as it won't have correctness issue. Here MAGIC_KEY is kind of ugly and it may lead to problems.

Good suggestio.

Reply:
> And why not use delete_cf for both branch?

This followers the pattern of `apply_delete`.